src/SqlVirtualMachine/SqlVirtualMachine/SqlVirtualMachine/Cmdlet/UpdateAzureSqlVM.cs

```diff
@@ -0,0 +1,225 @@
+        /// Offer of the new sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = NameParameterList,
+            HelpMessage = HelpMessages.OfferSqlVM)]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ResourceIdParameterList,
+            HelpMessage = HelpMessages.SkuSqlVM)]
+        [OfferCompleter]
+        public new string Offer { get; set; }
+
+        /// <summary>
+        /// Sku of the new sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = NameParameterList,
+            HelpMessage = HelpMessages.SkuSqlVM)]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ResourceIdParameterList,
+            HelpMessage = HelpMessages.SkuSqlVM)]
+        public new Sku? Sku { get; set; }
+
+        /// <summary>
+        /// SqlManagementType of the new sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = NameParameterList,
+            HelpMessage = HelpMessages.SqlManagementTypeSqlVM)]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ResourceIdParameterList,
+            HelpMessage = HelpMessages.SqlManagementTypeSqlVM)]
+        public new string SqlManagementType { get; set; } = "LightWeight";
+
+        /// <summary>
+        /// Tags will be associated to the new sql virtual machine
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = NameParameterList,
+            HelpMessage = HelpMessages.TagsSqlVM)]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ResourceIdParameterList,
+            HelpMessage = HelpMessages.TagsSqlVM)]
+        public new Hashtable Tags { get; set; }
+
+        /// <summary>
+        /// Sql virtual machine to be updated
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = NameInputObject,
+            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = true,
+            Position = 2,
+            HelpMessage = HelpMessages.InputObjectSqlVM)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = ResourceIdInputObject,
+            ValueFromPipelineByPropertyName = true,
+            ValueFromPipeline = true,
+            Position = 1,
+            HelpMessage = HelpMessages.InputObjectSqlVM)]
+        public AzureSqlVMModel SqlVM { get; set; }
+
+        /// <summary>
+        /// Resource id of the sql virtual machine that will be updated
+        /// </summary>
+        [Parameter(Mandatory = true,
+            ParameterSetName = ResourceIdParameterList,
+            Position = 0,
+            HelpMessage = HelpMessages.SqlVMResourceId)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = ResourceIdInputObject,
+            Position = 0,
+            HelpMessage = HelpMessages.SqlVMResourceId)]
```

Comment:
@rocchiele please add `ValueFromPipelineByPropertyName = true` to both parameter attributes for this parameter